docs/ASDK-2.0-Layout-Documention.md

```diff
@@ -0,0 +1,802 @@
+
+### Naming properties
+Certain properties have a different naming as on the web. For example `min-height` equivalent is the `minHeight` property. The full list of properties that control layout is documented in the `Layout Properties` section.
+
+### No margin / padding properties
+Layoutables don't have a padding or margin property. Instead wrapping a layoutable within a `ASInsetLayoutSpec` to apply padding or margin to the layoutable is the recommended way. See `ASInsetLayout` section for more information.
+
+### Missing features
+Certain features like `flexWrap` on a `ASStackLayoutSpec` are not supported currently. Please see the section for `Layout Properties` for the full list of properties that are supported.
+
+
+## Layoutables
+To understand AsyncDisplayKit's layout system it's crucial to understand what a layoutable and a layout spec is. Layoutables are classes that conform to the `ASLayoutable` protocol. The `ASLayoutable` protocol declares a methods for measuring the layout of an object. A layout is defined by an `ASLayout` return value, and must specify
+  1. the size (but not position) of the layoutable object and
+  2. the size and position of all of its immediate child objects.
+
+`ASDisplayNode` as well as `ASLayoutSpec` classes are conforming to the `ASLayoutable` protocol and provide this functionally out of the box. This allows you to compose nodes and layout specs to build very complex layouts in a declarative way.
+
+### Replaced Layoutables
+A replaced layoutable is any layoutable whose appearance and dimensions are defined by an external resource. Examples include `ASImageNode`, `ASTextNode`, `ASButtonNode` or `ASTextNode`. All other layoutables can be referred to as non-replaced layoutables. This would include `ASVideoNode`, `ASVideoPlayerNode`, `ASNetworkImageNode` or `ASEditableTextNode`.
+
+Replaced layoutables can have intrinsic dimensions—width and height values that are defined by the layoutable itself, rather than by its surroundings. For example, if an `ASImageNode` has a width set to `ASDimensionAuto`, the width of the linked image file will be used. For an `ASImageNode` the intrinsic content size would be calculated based on the text content.
+
+## Height and Width
+
+### The content area
+The content area is the area containing the real content of a layoutable. The properties `width`, `minWidth`, `maxWidth`, `height`, `minHeight` and `maxHeight` control the content size.
+
+### Setting a size
+A layoutable's height and width dimensions determine its size on the screen. The simplest way to set the dimensions of a layoutable is by setting the fixed `width` and `height` property. All dimensions in AsyncDisplayKit's layout system are unitless, and represent density-independent pixels. Setting dimensions this way is common for layoutables that should always render at exactly the same size, regardless of screen dimensions.
```

Comment:
`All dimensions in AsyncDisplayKit's layout system are unitless, and represent density-independent pixels. Setting dimensions this way is common for layoutables that should always render at exactly the same size, regardless of screen dimensions.`

Would suggest introducing the percent and point dimensions and go into the differences, rather than describing density-independent pixels. Namely, because developers will think they have already been working with density-independent pixels with UIKit. We should retrain their thinking when representing that you can not just have points, but also percents — which is why the layout system uses ASDimensions and not CGFloat.